core/server/models/session.js

```diff
@@ -0,0 +1,81 @@
+const ghostBookshelf = require('./base');
+
+const Session = ghostBookshelf.Model.extend({
+    tableName: 'sessions',
+
+    parse: function parse(attrs) {
+        attrs.session_data = JSON.parse(attrs.session_data);
+        return attrs;
+    },
+
+    format: function format(attrs) {
+        // CASE: format will be called when formatting all data for the DB
+        // including for SELECTs meaning that if we call findOne without
+        // a session_data property we'll get unintended JSON.stringify(undefined) calls
+        if (attrs.session_data) {
+            attrs.session_data = JSON.stringify(attrs.session_data);
+        }
+        return attrs;
+    },
+
+    user: function () {
+        return this.belongsTo('User');
+    }
+}, {
+
+    permittedOptions(methodName) {
+        const permittedOptions = ghostBookshelf.Model.permittedOptions.call(this, methodName);
+        if (methodName === 'upsert') {
+            return permittedOptions.concat('session_id');
+        }
+        if (methodName === 'destroy') {
+            return permittedOptions.concat('session_id');
+        }
+        return permittedOptions;
+    },
+
+    destroy(unfilteredOptions) {
+        if (unfilteredOptions.id) {
+            return ghostBookshelf.Model.destroy.call(this, unfilteredOptions);
+        }
+        const options = this.filterOptions(unfilteredOptions, 'destroy');
+
+        // Fetch the object before destroying it, so that the changed data is available to events
+        return this.forge({session_id: options.session_id})
```

Comment:
From https://github.com/TryGhost/Ghost/pull/9867/files#r219244147
Okay let's keep it like this, but we should keep in mind that we should make the destroy more flexible e.g. `options.destroyBy` or `(data, options)`. Not sure it's worth tracking it 🤔 There are some other places where we could use the base fn e.g. [here](https://github.com/TryGhost/Ghost/blob/2.1.4/core/server/models/tag.js#L113) (relations can be removed in event hooks). Maybe a grouped issue would be cool.

Reply:
I like `options.destroyBy` with a default of `id` - seems clean :+1:
Should I create an issue for this?

Reply:
Yeah 👍 Maybe you could mention that there are also other model implementations which can use the base model instead :) Thanks